modules/ContextUtils.js

```diff
@@ -13,61 +13,61 @@ function makeContextName(name) {
   return `@@contextSubscriber/${name}`
 }
 
-export function createContextProvider(name) {
+export function ContextProvider(name) {
   const contextName = makeContextName(name)
+  const listenersKey = `${contextName}/listeners`
```

Comment:
Why would there be name clashes? The `context` key in this case is going to be `@@contextSubscriber/router` – I fixed it from the original implementation where I munged the original context object unnecessarily.